pyccel/codegen/printing/ccode.py

```diff
@@ -1098,9 +1104,16 @@ def _cast_to(self, expr, dtype, precision):
 
     def _print_DottedVariable(self, expr):
         """convert dotted Variable to their C equivalent"""
+
+        lhs_code = self._print(expr.lhs)
+        name_code = self._print(expr.name)
+        code = f'{lhs_code}.{name_code}'
+        if self.stored_in_c_pointer(expr.lhs):
+            code = f'{self._print(ObjectAddress(expr.lhs))}->{name_code}'
```

Comment:
Personally I prefer the version with an if/else where `self._print(expr.lhs)`  is only calculated if it is used. Is there a reason why you prefer this version?

Reply:
no, but your suggestion is better.